src/common/mfu_daos.c

```diff
@@ -1881,3 +1901,2836 @@ int mfu_daos_flist_sync(
+                rc = 1;
+                goto out;
+            }
+            /* encode dataspace description
+             * in buffer then store in
+             * attribute on dataset
+             */
+            encode_buf = MFU_CALLOC(nalloc, sizeof(unsigned char));
+            if (encode_buf == NULL) {
+                rc = ENOMEM;
+                goto out;
+            }
+            status = H5Sencode1(hdf5->rx_dspace, encode_buf,
+                                &nalloc);
+            if (status < 0) {
+                MFU_LOG(MFU_LOG_ERR, "Failed to encode dataspace");
+                rc = 1;
+                goto out;
+            }
+            /* created attribute in HDF5 file with encoded
+             * dataspace for this record extent */
+            memset(attr_name, 0, ATTR_NAME_LEN);
+            memset(number_str, 0, ATTR_NAME_LEN);
+            memset(attr_num_str, 0, ATTR_NAME_LEN);
+            path_len = snprintf(number_str, ATTR_NAME_LEN, "%d",
+                                (int)(*ak_index));
+            if (path_len >= ATTR_NAME_LEN) {
+                MFU_LOG(MFU_LOG_ERR, "number_str is too long");
+                rc = 1;
+                goto out;
+            }
+            path_len = snprintf(attr_num_str, ATTR_NAME_LEN, "-%d", attr_num);
+            if (path_len >= ATTR_NAME_LEN) {
+                MFU_LOG(MFU_LOG_ERR, "attr number str is too long");
+                rc = 1;
+                goto out;
+            }
+            path_len = snprintf(attr_name, ATTR_NAME_LEN, "%s%s%s", "A-",
+                                number_str, attr_num_str);
+            if (path_len >= ATTR_NAME_LEN) {
+                MFU_LOG(MFU_LOG_ERR, "attr name is too long");
+                rc = 1;
+                goto out;
+            }
+            hdf5->attr_dims[0] = nalloc;
+            hdf5->attr_dspace = H5Screate_simple(1, hdf5->attr_dims,
+                                                 NULL);
+            if (hdf5->attr_dspace < 0) {
+                MFU_LOG(MFU_LOG_ERR, "failed to create attr");
+                rc = 1;
+                goto out;
+            }
+            hdf5->selection_attr = H5Acreate2(hdf5->rx_dset,
+                                              attr_name,
+                                              hdf5->rx_dtype,
+                                              hdf5->attr_dspace,
+                                              H5P_DEFAULT,
+                                              H5P_DEFAULT);
+            if (hdf5->selection_attr < 0) {
+                MFU_LOG(MFU_LOG_ERR, "failed to create selection attr");
+                rc = 1;
+                goto out;
+            }   
+            status = H5Awrite(hdf5->selection_attr, hdf5->rx_dtype,
+                              encode_buf);
+            if (status < 0) {
+                MFU_LOG(MFU_LOG_ERR, "failed to write attr");
+                rc = 1;
+                goto out;
```

Comment:
@dsikich Apparently, HDF5 has a limit of 2000 characters for metadata columns. I don't fully understand it, but this fails if roughly `size > 2000`. For now, we should probably enforce some limit on `size` above when we fetch. Maybe if `size > 2000` we can print and return an error.

Reply:
@daltonbohning hmm ok, I wonder if this would happen very often, were you testing with larger and larger sizes of recx?

Reply:
@dsikich Yeah, I was using recx sizes of 2049 and 4097